src/receive_imf.rs

```diff
@@ -1639,88 +1644,125 @@ async fn apply_group_changes(
         return Ok(None);
     }
 
-    let mut recreate_member_list = false;
     let mut send_event_chat_modified = false;
-
+    let mut removed_id = None;
     let mut better_msg = None;
-    let removed_id;
+    let mut apply_group_changes = false;
+
+    let mut recreate_member_list = if mime_parser
+        .get_header(HeaderDef::ChatGroupMemberRemoved)
+        .is_some()
+        || mime_parser
+            .get_header(HeaderDef::ChatGroupMemberAdded)
+            .is_some()
+    {
+        let self_added =
```

Comment:
Why is it a special case? If we missed a MemberAdded message for self, we won't be a group member, but other members will think that we are